### ISSUE TITLE
updated to 1.1 now showing as offline

### ISSUE THREAD

**Issue by CelticWebSolutions** [amiantos/pibar#31]
I've just updated from v1 to v1.1, v1 worked fine previously but v1.1 shows pinholes as offline, strangely when I go to preferences and test it shows as successful.  So I'm not quite sure what going on.

I'm on Catalina 10.15.4

Any suggestions of the best way to debug this please?

**Comment by amiantos** [comment]
You could try wiping all your settings and setting it back up and see if that makes a difference. Close PiBar, then open up terminal and type in `defaults delete net.amiantos.PiBar`. Re-open PiBar and set your Pi-hole back up. Let me know if that changes anything... if the tests are successful it _should_ be working...

**Comment by CelticWebSolutions** [comment]
Sorry for the delay, I'm about to give that a go, I also just purchased the AppStore release to support your development.

I do have one other minor issue that PiBar honest show up when I enabled Parallels toolbox "Hide Menu Icons" which is a handy add-on that allows you to basically have multiple tool bars by hiding some of the icons which can then be accessed by clicking a menu icon that swaps to all the hidden items.  Probably open another issue for that one so that they don't get mixed up.

**Comment by CelticWebSolutions** [comment]
I'm afraid that this suggestion didn't work, it definitely wiped the setting because I had to add my PiHoles again, added them in, they successfully tested, then it still shows as offline :(. any further suggestions?

All I see in Console is the following 

`default	12:04:38.114250+0100	PiBar	Task <7BC128DC-C058-4D01-8A8C-4C90178FB1CF>.<1> resuming, QOS(0x9) Voucher (null)
default	12:04:38.114723+0100	PiBar	Task <7BC128DC-C058-4D01-8A8C-4C90178FB1CF>.<1> summary for task success {transaction_duration_ms=0, response_status=404, cache_hit=1}
default	12:04:38.114871+0100	PiBar	Task <7BC128DC-C058-4D01-8A8C-4C90178FB1CF>.<1> finished successfully
`

**Comment by amiantos** [comment]
It's very weird that the test works but the actual data fetching doesn't work. If the test works, it should just work... I wonder what else could be unique about your setup that is causing this issue... I'm at a loss, especially since this only seems to be happening to you.

**Comment by amiantos** [comment]
I didn't ask before... what version of Pi-hole are you on?

**Comment by CelticWebSolutions** [comment]
V5.  I'm about to install an old version of PiBar to make sure it still works on the old version.  

You're welcome to remote access if you're that curious about it :)

**Comment by CelticWebSolutions** [comment]
OK so I just tried the original version.  It still didn't work after rI added my urls.  I did find one thing though, If I use the local address on the network, it works.  If however I used the remote address that I usually access it by, it has success in connecting but does not show as online.  So its' something to do with the way it connecting to remote addresses.  Either that or its something only network stopping it talking directly.  

So to be clear, I always connect to my PiHoles using pihole.domain.co.uk and pihole2.domain.co.uk, I use this for mobile apps (looking forward to yours!) and to access when I'm at home form a computer.  If I use the local network url, PiBar connects and successfully pulls information, if I use the remote url, it connects but then says it's offline.  I'm investigating to se if there's anything weird on my network stopping it, though I can't imagine what.

Only other detail I have for you is that one of my pinholes is running on an actual Pi running just pihole, the second is in a container on my Nas.

As mentioned, you are very welcome to remote access to run some tests if you wish.

Many thanks!
Huw

**Comment by amiantos** [comment]
That's interesting. When you're accessing through the domains are you using SSL? I assume you also have password protection turned on the Pi-hole interface too?

**Comment by CelticWebSolutions** [comment]
Yes, valid certificates and password protection.

**Comment by CelticWebSolutions** [comment]
Any other thoughts on this one?  Such as shame that I can no longer use it on my machine.

**Comment by amiantos** [comment]
Yeah I don’t know. I’m really at a loss, because the logic used for the connection test is the same logic used when actually fetching the pi-hole details. If the test works then the app should work fine... I can’t hazard a guess at the moment, sorry! While I work on the iOS version I’ll see if I can test out more situations and see what I come up with.

**Comment by CelticWebSolutions** [comment]
No worries.  Thanks anyway.

Good luck working on the IOS version.  Happy to be a tester as and when you need some :)

**Comment by tatablack** [comment]
For what it's worth, I'm having the same issue, only PiBar doesn't seem to work even when I use the IP address (i.e. it shows "Offline" even if "Test connection" succeeds).

Happy to help troubleshoot further if there are new ideas to try, or debugging levels to increase. 🙂

**Comment by ewrible** [comment]
Same issue, just started this morning.  No change in the PiBar version, nor the MacOS version (Ventura 13.1).  But I did do an update on the Pi-Hole side.  Was working just fine before the update to the latest on that end.  When I run the "test", says everything is fine, then shows "Initializing.." and then Offline.

**Comment by amiantos** [comment]
> Same issue, just started this morning. No change in the PiBar version, nor the MacOS version (Ventura 13.1). But I did do an update on the Pi-Hole side. Was working just fine before the update to the latest on that end. When I run the "test", says everything is fine, then shows "Initializing.." and then Offline.

@ewrible Looks like your issue was probably due to the latest Pi-hole update, PiBar 1.1.2 should fix this for you.

**Comment by amiantos** [comment]
Hoping this was fixed in v1.2/1.2.1

https://github.com/amiantos/pibar/releases/tag/macOS-1.2.1
